homeworks/feliistvanattila/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,108 @@
+import logging
+
+# Logging beállítása
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s - %(levelname)s - %(message)s',
+                    handlers=[
+                        # logging.FileHandler ("homeworks/feliistvanattila/4_exceptions_logging/task_manager.logger"),
+                        logging.FileHandler("task_manager.logger"),
+                        logging.StreamHandler()
+                    ])
+
+# Fájl neve, ahol a feladatokat tároljuk
+# TASK_FILE = "homeworks/feliistvanattila/4_exceptions_logging/tasks.txt"
+TASK_FILE = "tasks.txt"
+
+
+def read_tasks() -> list:
+    try:
+        with open(TASK_FILE, "r") as file:
+            tasks = file.readlines()
+            tasks = [task.strip() for task in tasks]
+        logging.info("Tasks read successfully.")
+        return tasks
+    except FileNotFoundError:
+        logging.error("The task file was not found.")
+        return []
+
+
+def add_task(task: str):
+    try:
+        with open(TASK_FILE, "a") as file:
+            file.write(task + "\n")
+        logging.info(f"Task '{task}' added successfully.")
+    except Exception as e:
+        logging.error("An error occurred while adding the task.")
+        logging.exception(e)
+
+# Feladat törlése a fájlból
+
+
+def remove_task(task: str):
+    tasks = read_tasks()
+    if task in tasks:
+        tasks.remove(task)
+        try:
+            with open(TASK_FILE, "w") as file:
+                for t in tasks:
+                    file.write(t + "\n")
+            logging.info(f"Task '{task}' removed successfully.")
+        except Exception as e:
+            logging.error("An error occurred while removing the task.")
+            logging.exception(e)
+    else:
+        logging.warning(f"Task '{task}' not found in the list.")
+
+
+def clear_all_task():
+    user_input = str(
+        input(f"Are you sure??? Press 'Y' to continue!: "))
+    if user_input == 'Y':
+        logging.info(f"Clearing task list started.")
+        for task in read_tasks():
```

Comment:
ezzel az a gond, hogy végigloopolsz egyesével az összes feladaton és minden egyes remove_task() hívásnál megnyitod a file-t majd bezárod (46-48. sor a kódodban). Ez ha nagyobb file-ról lenne szó, akkor nem túl effektív megoldás.
Ha fixen mindent törölni akarunk, akkor simán csak nyisd meg a file-t egyszer, törölj mindent és kész is vagyunk:

```
def clear_all_tasks():
    user_input = input("Are you sure? Press 'Y' to continue: ")
    if user_input == 'Y':
        try:
            with open(TASK_FILE, "w") as file:
                pass
            logging.info("All tasks cleared successfully.")
        except Exception as e:
            logging.error("An error occurred while clearing tasks.")
            logging.exception(e)
    else:
        logging.info("User canceled clearing all tasks.")
```
(átneveztem a function-öd többes számra, mivel itt legtöbb esetben egyszerre törölsz több taskot)
Ez a fenti azért működik, mert ha csak szimplán "w" write módban nyitod meg a file-t, egyből törlődik minden, mivel ez újraírást jelent.

Reply:
Szia István, köszi a visszajelzést. Igen nem volt hatékony, csak kicsit 'fürödtem' a kódújrahasznosításban :). Annyi volt bennem hogy a így törlöm akkor automatikusan nyoma is marad a törlésnek a logban. De persze átírtam beletettem azt amit javasoltál, mert úgy tényleg hatékonyabb.